source/common/quic/client_connection_factory_impl.cc

```diff
@@ -48,6 +48,14 @@ std::unique_ptr<Network::ClientConnection> createQuicNetworkConnection(
     }
   }
 
+  // Default enable RVCM connection option so that port migration is enabled.
+  quic::QuicTagVector connection_options;
+  if (info_impl->quic_config_.HasSendConnectionOptions()) {
+    connection_options = info_impl->quic_config_.SendConnectionOptions();
+  }
+  connection_options.push_back(quic::kRVCM);
+  info_impl->quic_config_.SetConnectionOptionsToSend(connection_options);
```

Comment:
I slightly prefer to either modifying the options in PersistentQuicInfoImpl constructor or duplicating the config and modifying the copy here. Mostly because it seems unnecessary for each connection creation to make the same modification to the shared config.

Reply:
Nice! Can you also change quic_http_integration_test not to set --quic_remove_connection_migration_connection_option?

Reply:
done.
The test quic_connection_persistent_info was not created in a kosher way.
So some more hacks were done.